### ISSUE TITLE
Problems with 'cordova platform add ios` – needs updating according to new apple standards

### ISSUE THREAD

**Issue by geoidesic** [apache/cordova-ios#1448]
# Bug Report

## Problem
Apple has updated minimum build version.
As well as various schema changes

### What is expected to happen?
- IPHONEOS_DEPLOYMENT_TARGET should be 12.0 by default for iOS `Cordova add platform iOS`
- Opening the project in Xcode should not cause warnings

### What does actually happen?
Build succeeds with warnings, warnings which will prevent the app from being shippable.

## Information
I suggest:
- create a new GitHub repo
- check it out
- create a new Cordova app `Cordova create`, add the iOS platform
- copy that into your GitHub repo folder
- commit that as your initial commit to enable yourself to get a diff
- then open that project in Xcode
- change the minimum `iOS Deployment Target` in both the project root folder and the app folder
- see the warning icon, click it to implement suggested changes
- now view the git diff to see how apple's expectations have changed for basic compatibility

<img width="804" alt="Screenshot 2024-07-18 at 12 00 31" src="https://github.com/user-attachments/assets/14aa9913-e363-4bca-928e-1aab6b3cd7f9">
<img width="1134" alt="Screenshot 2024-07-18 at 12 00 03" src="https://github.com/user-attachments/assets/1b300ff0-f718-40be-8604-30b222fb57e6">

### Command or Code
```
git clone git@github.com:your-account/cordova-test.git
cordova create cordovatest uk.co.domain.test cordova-test
mv -R cordovatest/ Cordova-test
cd cordova-test
git add *
git commit -m "initial commit"
git push
cordova platform add ios
cordova Build ios
```
That should allow you to see the errors. Then open the project in Xcode, click the warning icon and accept changes, change the build targets to the new minimum of 12, Then:

```
git diff
```

Which should show you what's changed in terms of apple's minimum build requirements.

### Environment, Platform, Device

iOS on a GitHub macOS-latest runner
and also iOS on my local Mac osx environment

### Version information

Mac OS X 14.5 (23F79)
Xcode Version 15.4 (15F31d)
cordova 12.0.0 (cordova-lib@12.0.1)
ios-deploy@1.12.2

## Checklist

- [x] I searched for existing GitHub issues
- [x] I updated all Cordova tooling to most recent version
- [x] I included all the necessary information above

**Comment by dpogue** [comment]
> Build succeeds with warnings, warnings which will prevent the app from being shippable.

It is only a warning, it does not impact the ability to submit the application to the App Store.

**Comment by geoidesic** [comment]
> It is only a warning, it does not impact the ability to submit the application to the App Store.

My experience was that in the Github Action's runner, this would cause an error and exit for the workflow.

**Comment by breautek** [comment]
Warnings isn't the cause of a non-zero exit code (which will trigger error routine), which can be observed by our CI: https://github.com/apache/cordova-ios/actions/workflows/ci.yml

It is possible that an actual error is buried in your build output somewhere. xcodebuild (like most other c compilers) does not stop when an error occurs, it will proceed to build every object file, and print information on the status of each compilation unit and then return an error code at the end.

In other words if the project has 1,000 source files and source file 1 has a build error, it will print the error details then proceed to build the other 999 source files (also printing information, burying the error). In this hypothetical scenario, the error would be relatively near the top, but of course in a real scenario errors could be anywhere in the build output.

Hope this helps.
